src/accelerate/accelerator.py

```diff
@@ -285,11 +287,18 @@ def __init__(
                 DeepSpeedPlugin() if os.environ.get("ACCELERATE_USE_DEEPSPEED", "false") == "true" else None
             )
         else:
-            assert isinstance(
-                deepspeed_plugin, DeepSpeedPlugin
-            ), "`deepspeed_plugin` must be an `accelerate.utils.DeepSpeedPlugin` object."
+            if isinstance(deepspeed_plugin, (tuple, list)):
+                for plugin in deepspeed_plugin:
+                    if not isinstance(plugin, DeepSpeedPlugin):
+                        raise TypeError("`deepspeed_plugin` must be a DeepSpeedPlugin object.")
+            else:
+                if not isinstance(deepspeed_plugin, DeepSpeedPlugin):
+                    raise TypeError("`deepspeed_plugin` must be a DeepSpeedPlugin object.")
+
+        if deepspeed_plugin is not None:
             os.environ["ACCELERATE_USE_DEEPSPEED"] = "true"  # use DeepSpeed if plugin is provided
-        if deepspeed_plugin:
+            if not isinstance(deepspeed_plugin, (tuple, list)):
+                deepspeed_plugin = [deepspeed_plugin]
```

Comment:
turn l286 into a list right away.

then move this code to line 290, and then you don't need to make the check if it's a plugin twice.

Reply:
possibly while at it rename to `deepspeed_plugins` - not sure if it breaks any public apis?

Reply:
It does/people tend to use `deepspeed_plugin`, however `.deepspeed_plugin` can refer to the first or active one and `.deepspeed_plugins` can be the full list.

Reply:
What you propose would be somewhat confusing but at the same time it's more backward compatible. But if it's well documented I think it works.

`.deepspeed_plugins` is unlikely to be used often if ever by end users.

But I'd ask for a 2nd opinion.

Reply:
Agreed, in this case (as I was still 50/50) it would *only* be the active plugin, while users can look in `accelerator.deepspeed_plugins` for all of them.

Reply:
Decided for this to be the active plugin, and then we have `deepspeed_plugins` aliased to grab them all